packages/perseus/src/perseus-types.ts

```diff
@@ -37,6 +37,14 @@ export type Size = [width: number, height: number];
 export type CollinearTuple = [Vector2, Vector2];
 export type ShowSolutions = "all" | "selected" | "none";
 
+// A utility type that constructs a widget map from a "registry interface".
+// The keys of the registry should be the widget type (aka, "categorizer" or
+// "radio", etc) and the value should be the option type stored in the value
+// of the map.
```

Comment:
Could you provide an example of the output from this function? Or an example usage?

Reply:
Yes. So I've started developing the idea of a type registry (because we have full widget options types, scoring data types, frontend rendering types). Each of these is map of widget type to the option type for that widget.

If you look at the usage in this PR, we use the `PerseusWidgetTypes` interface to create the `PerseusWidgetsMap` type. 

`PerseusWidgetTypes` looks like this (shortened a bit here): 

```typescript
export interface PerseusWidgetTypes {
    categorizer: CategorizerWidget;
    "cs-program": CSProgramWidget;
    definition: DefinitionWidget;
    ...
}
```

We use the helper: 

```typescript
export type PerseusWidgetsMap = MakeWidgetMap<PerseusWidgetTypes>;
```

And out comes the following type:

```typescript
type PerseusWidgetsMap = {
    [x: `categorizer ${number}`]: CategorizerWidget;
    [x: `cs-program ${number}`]: CSProgramWidget;
    [x: `definition ${number}`]: DefinitionWidget;
    ...
}
```

So you can think of this helper as a sort of type "generator". It takes a registry of widget types and generates a map type where the ID of the map is a widget id and the value is the correct option type.

Reply:
Thank you! I was thinking we could add an example to the comments, but that seems a bit long to add XD Thank you for writing that all out though. Helps to visualize it better ^_^